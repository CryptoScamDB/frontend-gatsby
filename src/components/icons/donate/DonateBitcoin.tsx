import React from 'react';
import Icon from '../Icon';
import LogoBitcoin from '../../../images/coins/bitcoin.png';

const DONATION_ADDR = {
    "ADDR": "32oirLEzZRhi33RCXDF9WHJjEb8RsrSss3",
    "HUMAN": "32oirLEzZRhi33RCXDF9WHJjEb8RsrSss3",
    "EXPLORER": "https://blockchain.info/address/32oirLEzZRhi33RCXDF9WHJjEb8RsrSss3"
}

export default () => (
    <a href={DONATION_ADDR.EXPLORER} target="_blank" rel="noopener noreferrer">
        <Icon 
            title={DONATION_ADDR.HUMAN} 
            src={LogoBitcoin} 
        />
    </a>
) as React.StatelessComponent;