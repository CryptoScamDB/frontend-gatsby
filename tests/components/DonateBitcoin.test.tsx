import React from 'react';
import renderer from 'react-test-renderer';
import DonateBitcoin from '../../src/components/icons/donate/DonateBitcoin';
import 'jest-styled-components';

describe('DonateBitcoin', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DonateBitcoin />).toJSON()!;
    const {
      props: { href }
    } = tree;
    expect(typeof href).toBe('string');
    expect(tree).toMatchSnapshot();
  });
});
