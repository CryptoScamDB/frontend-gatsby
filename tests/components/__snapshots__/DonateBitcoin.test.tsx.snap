// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateBitcoin renders correctly 1`] = `
.c0 {
  height: 26px;
  width: 26px;
}

<a
  href="https://blockchain.info/address/32oirLEzZRhi33RCXDF9WHJjEb8RsrSss3"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Donate Bitcoin"
    className="c0"
    src="test-file-stub"
    title="32oirLEzZRhi33RCXDF9WHJjEb8RsrSss3"
  />
</a>
`;
